src/docs/arch-snapshot.dox

```diff
@@ -172,13 +171,11 @@ This allowing a snapshot to ignore any uncommitted changes the checkpoint has wr
 
 @section snapshot_read_uncommitted Snapshot Read-uncommitted \& Modifies
 
-Snapshot read-uncommitted readers can not reconstruct a full value when reading an update chain that
-contains a modify. The reason is because another transaction with isolation snapshot can abort the
-modifies or updates in parallel to reader, which can unintentionally skip a required modify or
-we are not able to find a base update to apply the modifies on. Effectively read-uncommitted reader
-must read and race with the isolation snapshot write. It is difficult to distinguish if
-an aborted operation happened prior to the call of the function or if it has been done in parallel
-to the reader. Therefore whenever snapshot read-uncommitted readers encounter an update chain with a
-modify, the user will expect a WT_RESTART. Once the user reads the record again, the update
+It is possible that a read-uncommitted reader can not reconstruct a full value. This is
+because a snapshot isolation writer can abort the updates in parallel and leave the
+reader with an update list that does not contain enough information to reconstruct
+the full value. It is impossible to distinguish if an aborted modify or update happened prior
+to the call of the function, or if it is being done in parallel. In this case, we will return
+back to the user with a WT_RESTART error. Once the user reads the record again, the update
```

Comment:
This bit of documentation needs to be updated now that we've change this to `WT_ROLLBACK`.

Reply:
[cfd6173](https://github.com/wiredtiger/wiredtiger/pull/10311/commits/cfd6173e791bd0fb94c3b4c7af5a952e48394496)